Description: fix SA_SIGINFO usage to build on GNU/Hurd
 There is not SA_SIGINFO on GNU/Hurd, so do not try to use it.
Author: Pino Toscano <pino@kde.org>
Forwarded: http://bugreports.qt.nokia.com/browse/QTBUG-7805
--- a/src/testlib/qtestcase.cpp
+++ b/src/testlib/qtestcase.cpp
@@ -1521,7 +1521,11 @@ FatalSignalHandler::FatalSignalHandler()
 #ifndef Q_WS_QWS
         // Don't overwrite any non-default handlers
         // however, we need to replace the default QWS handlers
-        if (oldact.sa_flags & SA_SIGINFO || oldact.sa_handler != SIG_DFL) {
+        if (
+#ifdef SA_SIGINFO
+            oldact.sa_flags & SA_SIGINFO ||
+#endif
+            oldact.sa_handler != SIG_DFL) {
             sigaction(fatalSignals[i], &oldact, 0);
         } else
 #endif
